main.py

```diff
@@ -32,11 +30,39 @@
 RED = (255, 0, 0)
 
 def init_pygame():
-    width, height = 1000, 800
+    width, height = 800, 600
     screen = pygame.display.set_mode((width, height))
     pygame.display.set_caption("HackerHunt")
     return screen, width, height
 
+def read_high_scores():
+    try:
+        with open("high_scores.txt", "r") as file:
+            scores = [int(line.strip()) for line in file]
+        return scores
+    except FileNotFoundError:
+        return []
+
+def write_high_score(score):
+    scores = read_high_scores()
+    scores.append(score)
+    scores = sorted(scores, reverse=True)[:5]  # Keep only top 5 scores
+    with open("high_scores.txt", "w") as file:
+        for score in scores:
+            file.write(f"{score}\n")
+
+def get_high_score():
+    scores = read_high_scores()
+    return max(scores) if scores else 0
+def draw_high_scores(screen):
+    high_scores = read_high_scores()
+    screen.fill(BLACK)
+    title = font.render("High Scores", True, WHITE)
+    screen.blit(title, (350, 100))
+    for index, score in enumerate(high_scores):
+        text = font.render(f"{index + 1}. {score}", True, WHITE)
+        screen.blit(text, (350, 150 + 40 * index))
+
 def ask_question_with_node_class(node):
```

Comment:
here we should re-write this function like this:

```
def ask_question_with_node_class(node):
     challenge = next((problem for problem in PROBLEMS if problem.id == node.id), None)
    if challenge:
        challenge.run()
        return challenge.is_completed()
    print("No valid challenge found for ID:", node.id)
    return True

Reply:
this way we check if a valid challenge associated with that node exists then we return its completed state otherwise return default True